config/src/parameters/user.rs

```diff
@@ -347,38 +350,36 @@ pub(crate) fn private_key_from_env<E: Error>(
 #[derive(Debug)]
 pub struct Genesis {
     pub public_key: PublicKey,
-    pub private_key: Option<PrivateKey>,
     pub file: Option<PathBuf>,
+    pub signature: Option<String>,
```

Comment:
Please use `GenesisSignature` directly, or a type that wraps it. Just `String` is way less useful.

It is as if `public_key` would be just some `String` and not `PublicKey`. Will require more parsing steps further, while it could be done right here, seamlessly, by `serde`.